fw/apm.c

```diff
@@ -1714,13 +1717,17 @@ static int
 tfw_apm_cfgend(void)
 {
 	int r;
+	TfwApmData *r2;
 
 	if (tfw_runstate_is_reconfig())
 		return 0;
 
-	tfw_apm_global_data = tfw_apm_data_create();
-	if (!tfw_apm_global_data)
-		return -EINVAL;
+	r2 = tfw_apm_data_create();
+	if (IS_ERR(r2))
+		return PTR_ERR(r2);
+	else
```

Comment:
There's no need for `else`. Just continue the flow of the code.

Reply:
fixed